lib/rules/id-length.js

```diff
@@ -28,7 +30,7 @@ module.exports = function(context) {
                 // regular property assignment
                 parent.parent.left === parent || (
                     // or the last identifier in an ObjectPattern destructuring
-                    parent.parent.type === "Property" && parent.parent.value === parent &&
+                    parent.parent.type === "Property" && properties && parent.parent.value === parent &&
```

Comment:
> overriding the method with false as I suggested below for Property may work for this too

There are many conditions evaluated here and I must admit I wasn't sure of all the cases it covered, I wrote my failing tests first and just adding that condition made them pass so I kept it to that, rather than going for a top-down approach. If you say bumping that check up doesn't create false positives/negatives then I can do that!